Sources/FireKit/UserSessionManager.swift

```diff
@@ -9,224 +9,232 @@ import Foundation
+            // User found -- call completion to handle any post-reauth workflow
+            onAuthSessionStart(user: user, error: nil)
+        } else if defaultToAnonymousUser {
+            print("PTUserSessionManager: Auth State Changed -- No User (Anon)")
+            // No user logged in and we want to default to anonymous user. Start an Anonymous User Session
+            startAnonymousSession()
+        } else if state == .authenticated {
+            print("PTUserSessionManager: Auth State Changed -- No User (Signout)")
+            // A user just signed out (i.e. we've gone from user to no user)
+            state = .unauthenticated
+            userID = nil
+            onAuthSessionEnd()
         } else {
-          completion?(.failure(PTUserSessionError.unknownAuthError))
+            print("PTUserSessionManager: Auth State Changed -- No User")
+            state = .unauthenticated
+            userID = nil
         }
-      }
-    }
-  }
-  
-  open func login(email: String, password: String, completion: ((Result<FirebaseAuth.User, Error>) -> Void)? = nil) {
-    print("PTUserSessionManager: Logging In")
-    let oldSession = Auth.auth().currentUser
-    Auth.auth().signIn(withEmail: email, password: password) { result, error in
-      if let error = error {
-        print("PTUserSessionManager: Logging In (Error)")
-        print(#fileID, #line, #function, error)
-        completion?(.failure(error))
-      }
-      else if let user = result?.user {
-        self.userID = user.uid
-        print("PTUserSessionManager: Logging In (Success)")
-        
-        // Delete old anon session if we login with an existing non-anon session
-        if oldSession?.isAnonymous == true {
-          oldSession?.delete { error in
-            print("PTUserSessionManager: Logging In (Deleted Anon)")
+    }
+    
+    open func startAuthListener() {
+        print("PTUserSessionManager: Auth Listener Started")
+        Auth.auth().addStateDidChangeListener(onAuthStateChanged)
+    }
+    
+    // MARK: - Password Changes
+    
+    open func changePasswordRequest(replace oldPassword: String, with newPassword: String, completion: ((Result<Void, Error>) -> Void)? = nil) {
+        let user = Auth.auth().currentUser
+        let credentials = EmailAuthProvider.credential(withEmail: email, password: oldPassword)
+        user?.reauthenticate(with: credentials, completion: { (result, error) in
+            if let error = error {
+                completion?(.failure(error))
+            } else {
+                // User re-authenticated.
+                user?.updatePassword(to: newPassword){ error in
+                    if let error = error { completion?(.failure(error)) }
+                    else { completion?(.success(())) }
+                }
+            }
+        })
+    }
+    
+    open func forgetPassword(email: String, completion: ((Result<Void, Error>) -> Void)? = nil) {
```

Comment:
bence bu functionların override edilmesini istemiyoruz sanki, completion'ı custom bi function olanlar sadece override edilebilmeli dememişmiydik?

Reply:
why not? belki forgotpassword icin Firebase emailini kullanmak istemiyo adam?

Reply:
how does that make more sense? override edip email input'u alan yeni bi function yazıp, email'ı kullanmicak diosun?

Reply:
emaili niye kullanmasin anlamadim? Adam app yazdi, forgot password flowu firebase uzerinden degil baska bir email client uzerinden yapmak istedi diyorum postmark sendgrid, etc.